tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/action/WatchWorkspaceModulesAction.java

```diff
@@ -123,17 +128,34 @@ else if ("stop".equals(_action)) {
 
 						Set<IProject> childProjects = liferayGradleWorkspaceProject.getChildProjects();
 
-						for (IProject childProject : childProjects) {
-							_undeployModuleFromServer(childProject);
-						}
+						childProjects.stream(
+						).map(
+							childProject -> LiferayCore.create(IBundleProject.class, childProject)
+						).filter(
+							Objects::nonNull
+						).parallel(
+						).forEach(
+							bundleProject -> {
+								IProject childProject = bundleProject.getProject();
+
+								new StopWatchBundleJob(
```

Comment:
Let's not create a single stop watch bundle job for each project.  But instead let's collect all of the projects that need to have the bundle uninstalled and pass to the stop watch bundle job instead.

Reply:
ok